assets/admin/editor-wizard/editor-wizard-modal.js

```diff
@@ -94,12 +94,25 @@ const EditorWizardModal = () => {
 
 	// Choose steps by post type.
 	const stepsByPostType = {
-		course: [ CourseDetailsStep, UpgradeStep, CoursePatternsStep ],
+		course: [ CourseDetailsStep, CourseUpgradeStep, CoursePatternsStep ],
 		lesson: [ LessonDetailsStep, LessonPatternsStep ],
 	};
-	const { postType } = useSelect( ( select ) => ( {
-		postType: select( editorStore )?.getCurrentPostType(),
-	} ) );
+	const { postType, senseiProExtension } = useSelect(
+		( select ) => ( {
+			postType: select( editorStore )?.getCurrentPostType(),
+			senseiProExtension: select(
+				EXTENSIONS_STORE
+			).getSenseiProExtension(),
+		} ),
+		[]
+	);
+
+	if ( ! senseiProExtension || senseiProExtension.is_installed === true ) {
+		stepsByPostType.course = stepsByPostType.course.filter(
+			( step ) => step !== CourseUpgradeStep
+		);
+	}
```

Comment:
This is making the Upgrade step flash in/out if you go over the first step somewhat fast.
https://user-images.githubusercontent.com/799065/170244529-f19593a5-145b-42e3-baa5-b493d77157c7.mp4

Any idea on how to workaround that?